vllm/worker/worker_base.py

```diff
@@ -215,6 +216,20 @@ def execute_worker(self, worker_input: WorkerInput) -> None:
     def execute_model(
         self,
         execute_model_req: Optional[ExecuteModelRequest] = None
+    ) -> Optional[List[SamplerOutput]]:
+        """Executes at least one model step on the given sequences, unless no
+        sequences are provided."""
+        if self.use_spmd_worker:
+            assert execute_model_req is not None, (
+                "VLLM_USE_SPMD_WORKER=1 requires each worker to take in an "
+                "ExecuteModelRequest")
+            return self._execute_model_spmd(execute_model_req)
+
+        return self._execute_model_with_nccl_control_plane(execute_model_req)
+
+    def _execute_model_with_nccl_control_plane(
```

Comment:
Can you update the docstring

"""
This implementation uses nccl broadcasting primitive to broadcast input data to all workers
"""